tree/ntuple/v7/inc/ROOT/REntry.hxx

```diff
@@ -86,54 +86,87 @@ class REntry {
 public:
    using ConstIterator_t = decltype(fValues)::const_iterator;
 
+   /// The field token identifies a top-level field in this entry. It can be used for fast indexing in REntry's
+   /// methods, e.g. BindValue
+   class RFieldToken {
+      friend class REntry;
+      std::size_t fIndex;     ///< the index in fValues that belongs to the top-level field
+      std::uint64_t fModelId; ///< Safety check to prevent tokens from other models being used
+      RFieldToken(std::size_t index, std::uint64_t modelId) : fIndex(index), fModelId(modelId) {}
+   };
+
    REntry(const REntry &other) = delete;
    REntry &operator=(const REntry &other) = delete;
    REntry(REntry &&other) = default;
    REntry &operator=(REntry &&other) = default;
    ~REntry() = default;
 
+   /// The ordinal of the top-level field fieldName; can be used in other methods to address the corresponding value
+   RFieldToken GetToken(std::string_view fieldName) const
+   {
+      for (std::size_t i = 0; i < fValues.size(); ++i) {
+         if (fValues[i].GetField().GetFieldName() == fieldName)
+            return RFieldToken(i, fModelId);
+      }
+      throw RException(R__FAIL("invalid field name: " + std::string(fieldName)));
```

Comment:
I'd suggest a slight reordering of the logic to avoid having the compiler possibly complain about no `return` at the end of the function.
```C++
     auto it = ....;
     if ( it == fValues.end() ) {
           throw RException(...);
     }
     return RFieldToken(...);
```